src/interface.ts

```diff
@@ -89,16 +96,33 @@ interface DesktopAgent {
    * Returns a promise that resolves to an Array. The resolved dataset & metadata is Desktop Agent-specific.
    * If intent argument is falsey, then all possible intents - and apps corresponding to the intents - are resolved for the provided context.
    * If the resolution errors, it returns an `Error` with a string from the `ResolveError` enumeration.
+   * 
+   * ```javascript
+   * //find what Apps can support SartChat with the given context structure
+   * var intentR = await agent.resolve("StartChat", context);
+   * 
+   * //find what Intents and Apps are supported for a given context
+   * var actionR = await agent.resolve(false, context);
```

Comment:
This should be:
`var actionMetadata = await agent.resolve(null, context)`
I don't think using `false` is a good idea, it is misleading.

Also, let's not use abbreviations for these examples, and write out what the variables represent.

Reply:
I would extend like so:
```javascript
// find what intents and apps are supported for a given context
var actionMetadata = await agent.resolve(null, context);
// e.g.:
// [{
//     intent: { name: "StartCall", displayName: "Call" },
//     apps: [{ name: "Skype" }]
// },
// {
//     intent: { name: "StartChat", displayName: "Chat" },
//     apps: [{ name: "Skype" }, { name: "Symphony" }, { name: "Slack" }]
// }];

// select a particular intent and app to raise 
const selectedAction = actionMetadata[1];
const selectedApp = selectedAction.apps[0];
await agent.raiseIntent(selectedAction.intentName, context, selectedApp.name);
```